apisix/plugins/jwt-auth.lua

```diff
@@ -235,15 +230,10 @@ local function get_rsa_or_ecdsa_keypair(conf)
     local public_key = conf.public_key
     local private_key = conf.private_key
 
-    if public_key and private_key then
-        return public_key, private_key
-    elseif public_key and not private_key then
-        return nil, nil, "missing private key"
-    elseif not public_key and private_key then
+    if not public_key then
         return nil, nil, "missing public key"
-    else
-        return nil, nil, "public and private keys are missing"
     end
+    return public_key, private_key
```

Comment:
Why is there still a `private_key`?

Reply:
To facilitate generating jwt tokens in tests, I added a `_M.gen_token` method. In the test, I can use `local gen_token = require("apisix.plugins.jwt-auth").gen_token`. `gen_token` depends on this function, so there is still a private_key.

Reply:
I think you need to completely distinguish between verifying the legality of the token in client requests and `gen_token` in tests, otherwise this function makes it very misleading that a `private_key` field is still needed in the `jwt-auth` configuration.